# Pruebas de Hipótesis 2

Exploraremos diferentes tipos de pruebas de hipótesis, incluyendo pruebas de una cola y dos colas, utilizando diferentes distribuciones. Además, interpretaremos los valores p y los estadísticos calculados en cada caso.

## Repaso
Las pruebas de hipótesis son una herramienta estadística fundamental para tomar decisiones basadas en datos. Nos permiten evaluar si existe suficiente evidencia en una muestra para inferir algo sobre una población.

## Prueba t de una muestra (dos colas)

### Situación
Supongamos que somos una compañía que produce un tipo de cable que debe soportar una tensión promedio de 500 kg antes de romperse. Queremos saber si un nuevo proceso de fabricación ha cambiado la resistencia promedio del cable.

### Planteamiento de Hipótesis

Hipótesis nula (H₀): La resistencia promedio es igual a 500 kg (μ = 500).

Hipótesis alternativa (H₁): La resistencia promedio es diferente de 500 kg (μ ≠ 500).

### Datos sintéticos

En vez de tomar datos de algun archivo o base, generaremos nuestros propios datos.
Esto es una práctica común en la industria ya que muchas veces no contaremos con suficientes datos "originales" para entrenar modelos o para hacer pruebas estadísticas. 

Para generar buenos datos sintéicos existen diferentes técnicas. En caso de conocer los datos, se les puede hacer algunas modificaciones a los datos para generar nuevo contenido. Por ejemplo, si tenémos diferentes imágenes de gatos y queremos hacer un clasificador, es posible construir datos adicionales al girar, invertir e incluso aplicar filtros a las fotos originales. 
En caso de no contar con datos reales, podemos generar datos sintéticos con fines de hacer ejercicios y explorar escenarios posibles. Sin embargo, recordemos que en la práctica se deberán contar con datos reales para validar los sintéticos y no caer en sesgos.

Generemos entonces un conjunto de datos que simulen tener las condiciones planteadas en la situación.

In [2]:
import numpy as np
from scipy import stats

In [30]:
# Generamos una muestra de resistencias
np.random.seed(42)  # Para reproducibilidad
muestra = np.random.normal(loc=505, scale=15, size=30)  # Supongamos una media real 505 kg y std de 15
muestra

array([512.4507123 , 502.92603548, 514.71532807, 527.84544785,
       501.48769938, 501.48794565, 528.68819223, 516.51152094,
       497.95788421, 513.13840065, 498.04873461, 498.0140537 ,
       508.62943407, 476.30079633, 479.12623251, 496.56568706,
       489.80753319, 509.71370999, 491.37963887, 483.81544448,
       526.98473153, 501.61335549, 506.01292307, 483.62877721,
       496.83425913, 506.66383885, 487.73509634, 510.63547028,
       495.99041965, 500.62459375])

### Prueba de hipótesis

Se utiliza una prueba **t de Student** porque estamos comparando la media de una muestra (las resistencias de los cables) con un valor específico de referencia (500 kg) y queremos evaluar si hay una diferencia significativa entre la media de la muestra y ese valor. Esta prueba es apropiada cuando:

    1. El tamaño de la muestra es relativamente pequeño (en este caso, 30 observaciones), y no se conoce la desviación estándar de la población, por lo que necesitamos estimarla a partir de la muestra.
    
    2. Se asume que la variable que estamos evaluando sigue una distribución aproximadamente normal (la resistencia de los cables sigue una distribución normal en la mayoría de los procesos de manufactura).
    
La prueba t de Student es ideal en estas situaciones, pues nos permite comparar si la media muestral difiere significativamente de un valor hipotético (en este caso, 500 kg) cuando la varianza es desconocida y debe ser estimada a partir de los datos.

### Pruebas de una o dos colas

Podemos preguntarnos ¿Es una prueba de una cola o de dos colas?

La diferencia entre una prueba de una o dos colas radica en las hipótesis que hagamos y fundamentalmente en lo que querramos detectar o probar. 

    Prueba de dos colas: Se utiliza cuando queremos detectar si la media es diferente (ya sea mayor o menor) de un valor específico, sin importar la dirección del cambio.

    Prueba de una cola: Se utilizaría si tuviéramos una hipótesis alternativa específica, como si quisiéramos evaluar solo si la resistencia ha aumentado o disminuido, pero no ambas.

Nuestro caso cae en una prueba de una cola ya que lo que nos interesa es saber si la resistencia promedio del cable es diferente de 500 kg. No nos interesa, en este caso, considerar que pueda resistir mas o menos queremos exactamente el peso de 500 kg. Por eso el valor p se calcula para ambos extremos de la distribución t.

In [32]:
# Realizamos la prueba t de una muestra
t_statistic, p_value = stats.ttest_1samp(muestra, popmean=500)

print(f'Estadístico t: {t_statistic:.2f}')
print(f'Valor p: {p_value:.4f}')

Estadístico t: 0.88
Valor p: 0.3842


### Interpretación y Decisión

Estadístico t: Mide cuántas desviaciones estándar está la media muestral de la media poblacional bajo H₀.

Valor p: Probabilidad de obtener un resultado tan extremo como el observado, asumiendo que H₀ es verdadera.

In [26]:
alpha = 0.05  # Nivel de significancia

if p_value < alpha:
    print("Rechazamos H₀: La resistencia promedio es significativamente diferente de 500 kg.")
else:
    print("No rechazamos H₀: No hay evidencia suficiente para decir que la resistencia promedio es diferente de 500 kg.")


No rechazamos H₀: No hay evidencia suficiente para decir que la resistencia promedio es diferente de 500 kg.


## Prueba t de una muestra (una cola)

### Situación

Ahora, queremos saber si el nuevo proceso de fabricación ha aumentado la resistencia promedio del cable.

### Hipótesis

Hipótesis nula (H₀): La resistencia promedio es menor o igual a 500 kg (μ ≤ 500).

Hipótesis alternativa (H₁): La resistencia promedio es mayor a 500 kg (μ > 500).

### Pruebas de una o dos colas

Con el criterio anterio, podemos detectar con facilidad si nuestro caso en una prueba de una o dos colas.

En este caso, nos interesa probar si la media de la resistencia del cable ha aumentado. Es decir, si el nuevo proceso de fabricación ha mejorado la resistencia. No estamos interesados en la posibilidad de que la resistencia sea menor a 500 kg.

La hipótesis alternativa (H₁) plantea que la media es **mayor** que 500 kg, lo que significa que estamos enfocados solo en un lado (la "cola" derecha) de la distribución de la prueba t.

### ¿Qué pasa con el p-value?

Cuando se utiliza una prueba de dos colas, el valor p mide la probabilidad de que los resultados sean más extremos que los observados, tanto en el lado positivo como en el negativo de la distribución t.

En una prueba de dos colas, estamos interesados en desviaciones en ambos sentidos: tanto si la media es mayor como si es menor que el valor de la hipótesis nula.

En una prueba de una cola, como solo estamos interesados en una dirección (si la resistencia es mayor que 500 kg), el valor p de la prueba de dos colas debe dividirse por 2. Esto es porque en la prueba de dos colas, el valor p refleja la probabilidad de obtener resultados extremos en ambos extremos (positivo y negativo), pero en una prueba de una cola, solo nos importa el extremo que nos interesa (en este caso, el extremo derecho, que corresponde a valores mayores).

In [28]:
# Valor p ajustado para una cola
p_value_one_tailed = p_value / 2

print(f'Valor p (una cola): {p_value_one_tailed:.4f}')

if (t_statistic > 0) and (p_value_one_tailed < alpha):
    print("Rechazamos H₀: La resistencia promedio es significativamente mayor a 500 kg.")
else:
    print("No rechazamos H₀: La resistencia promedio es menor o igual a 500 Kg.")


Valor p (una cola): 0.1921
No rechazamos H₀: La resistencia promedio es menor o igual a 500 Kg.


##  Prueba t para muestras independientes (dos colas)

### Situación
Queremos comparar la resistencia promedio de cables fabricados con dos materiales diferentes.

### Datos sintéticos

In [34]:
# Generamos datos para dos grupos
material_A = np.random.normal(500, 15, 30)
material_B = np.random.normal(510, 15, 30)

### Hipótesis

Hipótesis nula (H₀): Las resistencias promedio son iguales (μ₁ = μ₂).

Hipótesis alternativa (H₁): Las resistencias promedio son diferentes (μ₁ ≠ μ₂).

### Prueba de hipótesis

Recordemos que consideraremos una prueba de dos colas porque no estamos interesados en una dirección específica del cambio (mayor o menor), sino simplemente en cualquier diferencia significativa entre los dos grupos.

En otras palabras, **no estamos haciendo ninguna suposición sobre qué grupo tendría una resistencia promedio mayor** solo queremos ver si hay alguna diferencia. Esto hace que la prueba sea bidireccional, o de dos colas.

In [40]:
t_statistic, p_value = stats.ttest_ind(material_A, material_B)

print(f'Estadístico t: {t_statistic:.2f}')
print(f'Valor p: {p_value:.4f}')

Estadístico t: -3.22
Valor p: 0.0021


### Interpretación y Decisión

### Interpretación del estadístico t

El estadístico t mide la diferencia entre las medias de los dos grupos en términos de las desviaciones estándar combinadas (o estándar del error). En este caso, el valor negativo de t indica que la media de la resistencia del Material A es menor que la del Material B.

Un estadístico t de -3.22 significa que la media de la resistencia del Material A está 3.22 desviaciones estándar por debajo de la media de la resistencia del Material B. Es una diferencia bastante grande en términos estadísticos.

In [43]:
if p_value < alpha:
    print("Rechazamos H₀: Las resistencias promedio son significativamente diferentes.")
else:
    print("No rechazamos H₀: No hay evidencia suficiente para decir que las resistencias promedio son diferentes.")


Rechazamos H₀: Las resistencias promedio son significativamente diferentes.


## ANOVA de un factor (Análisis de Varianza)

### Situación
Queremos comparar el efecto de tres dietas diferentes en el aumento de peso de animales de laboratorio.

### Datos sintéticos

In [103]:
# Generamos datos para tres dietas
dieta_1 = np.random.normal(7, 1, 20)
dieta_2 = np.random.normal(7, 1, 20)
dieta_3 = np.random.normal(7, 2, 20)

### Hipótesis

Hipótesis nula (H₀): Las medias de las tres dietas son iguales.

Hipótesis alternativa (H₁): Al menos una de las medias es diferente.

### Prueba de hipótesis

In [107]:
f_statistic, p_value = stats.f_oneway(dieta_1, dieta_2, dieta_3)

print(f'Estadístico F: {f_statistic:.2f}')
print(f'Valor p: {p_value:.4f}')

Estadístico F: 0.10
Valor p: 0.9022


### Interpretación y decisión

El estadístico F mide la relación entre la **variabilidad entre grupos** (es decir, cómo difieren las medias de las dietas) y la **variabilidad dentro de los grupos** (es decir, la variabilidad en los aumentos de peso dentro de cada dieta). Un valor de F alto sugiere que hay más variabilidad entre los grupos que dentro de los grupos, lo cual es una señal de que las medias de los grupos podrían no ser iguales.

In [110]:
if p_value < alpha:
    print("Rechazamos H₀: Hay diferencias significativas entre las dietas.")
else:
    print("No rechazamos H₀: No hay evidencia suficiente para decir que existen diferencias entre las dietas.")


No rechazamos H₀: No hay evidencia suficiente para decir que existen diferencias entre las dietas.
